x-pack/plugins/apm/common/utils/join_by_key/index.ts

```diff
@@ -56,16 +57,16 @@ export function joinByKey(
     merge({}, a, b)
 ) {
   const keys = castArray(key);
+  const map = new Map();
   return items.reduce<Array<Record<string, any>>>((prev, current) => {
-    let item = prev.find((prevItem) =>
-      keys.every((k) => isEqual(prevItem[k], current[k]))
-    );
+    const stableKey = stableStringify(keys.map((k) => current[k]));
```

Comment:
Can you add a comment explaining why this is needed?

Reply:
Done.